app/controllers/concerns/basic_authenticator.rb

```diff
@@ -2,19 +2,52 @@
 
 module BasicAuthenticator
   extend ActiveSupport::Concern
+  include Authenticators
   
   included do
     include ActionController::HttpAuthentication::Basic::ControllerMethods
   end
   
   def perform_basic_authn
     authenticate_with_http_basic do |username, password|
-      credentials = Credentials[Role.roleid_from_username(account, username)]
-      if credentials && credentials.authenticate(password)
-        authentication.authenticated_role = credentials.role
+      authenticator_login(username, password).tap do |response|
+        authentication.authenticated_role = ::Role[response.role_id]
         authentication.basic_user = true
       end
+    rescue ::Authentication::Strategy::InvalidCredentials, 
+           ::Authentication::Strategy::InvalidOrigin,
+           ::Authentication::Security::NotAuthorizedInConjur
```

Comment:
Per Jonah: `InvalidOrigin` and `NotAuthorized` could return a more useful message with the `Unathorized` response